policies/policies.go

```diff
@@ -117,64 +147,122 @@
 	return data, nil
 }
 
-func RunMap(baseDir string, metric string, m map[string]interface{}, holder MetricConfigurationSource) (result *Result, err error) {
+type queryCache struct {
```

Comment:
Normally, I would expect that all types, structs and global variables are placed at the top of the code before all the methods and functions (It helps to read the code step by step, though)